rust/processor/src/db/postgres/migrations/2024-08-14-015329_add_daily_volume_and_tvl_growth/up.sql

```diff
@@ -0,0 +1,89 @@
+-- Your SQL goes here
+
+CREATE OR REPLACE FUNCTION assert_arrays_equal_length(BIGINT[], NUMERIC[]) RETURNS void AS $$
+BEGIN
+    IF array_length($1, 1) != array_length($2, 1) THEN
+        RAISE EXCEPTION 'Arrays are not of equal length';
+    END IF;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION assert_arrays_equal_length(BIGINT[], BIGINT[]) RETURNS void AS $$
+BEGIN
+    IF array_length($1, 1) != array_length($2, 1) THEN
+        RAISE EXCEPTION 'Arrays are not of equal length';
+    END IF;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION one_day_ago_micros() RETURNS BIGINT AS $$
+    SELECT (EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) * 1000000 - 86400000000)::BIGINT;
```

Comment:
Addressed in 0db80eb, I changed `BIGINT` to timestamp in the table and in the code, I now compare against `- 24 hours`, noted in the code why but here's the explanation:

```sql
-- Note that we use `INTERVAL '24' HOUR` because `'1' DAY` does not always equal 24h.
-- See: https://www.postgresql.org/docs/9.1/functions-datetime.html, quote below:
-- | this means interval '1 day' does not necessarily equal interval '24 hours'.
```